File: typescript/src/main/vsdk/toolkit/common/linealAlgebra/Ray.ts
import { Vector3D } from "./Vector3D";

export class Ray {
  public position: Vector3D;
  public direction: Vector3D;

  public constructor(position?: Vector3D, direction?: Vector3D) {
    this.position = position ?? new Vector3D();
    this.direction = direction ?? new Vector3D();
  }
}

File: java/base/src/main/java/vsdk/toolkit/common/linealAlgebra/Ray.java
package vsdk.toolkit.common.linealAlgebra;

// Direction should be normalized

public class Ray {
    public Vector3D position;
    public Vector3D direction;

    public Ray() {
        position = new Vector3D();
        direction = new Vector3D();
    }

    public Ray(Vector3D position, Vector3D direction) {
        this.position = position;
        this.direction = direction;
    }
}
